### PR TITLE
Infra: Add CI

### PR DESCRIPTION
**Pull request by ryan-roemer**
Adds `pnpm` scenario CI. Just build for now.

### CODE REVIEW

**Review by gksander — APPROVED** [review]
I like it!

File: .github/workflows/pnpm-ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - uses: pnpm/action-setup@v2.2.2
+        with:
+          version: 7
+
+      - name: Get pnpm store directory
+        id: pnpm-cache
+        run: echo "::set-output name=pnpm_cache_dir::$(pnpm store path)"
```
Comment (scottrippey) [inline .github/workflows/pnpm-ci.yml:38]: This is crazy, very cool!

File: .github/workflows/pnpm-ci.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+      - name: Setup pnpm cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.pnpm-cache.outputs.pnpm_cache_dir }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
```
Comment (scottrippey) [inline .github/workflows/pnpm-ci.yml:44]: Super cool too!

**Review by scottrippey — COMMENTED** [review]
Very cool Actions setup!  Love the cache.

**Pull request merged**